lib/bulk_consumer.rb

```diff
@@ -0,0 +1,19 @@
+module GovukMessageQueueConsumer
+  # FIXME: this is a proof of concept for a consumer that uses self-declared,
+  # transient queues. We should adapt the consumer in the gem if this works.
+  class BulkConsumer < Consumer
+    NUMBER_OF_MESSAGES_TO_PREFETCH = 10
+
+  private
+
+    def queue
+      @queue ||= begin
+        channel.prefetch(NUMBER_OF_MESSAGES_TO_PREFETCH)
+
+        channel.queue(@queue_name, auto_delete: true, durable: false).tap do |queue|
```

Comment:
I'm just trying to wrap my head around `auto_delete` from the docs, so this question might be a bit confused. 🤔 

Does `auto_delete: true` mean that the queue would be deleted if the consumer goes away, e.g. if rummager was redeployed in the middle of a bulk indexing?

Is that the intended behaviour, and does the queue get recreated if a new message with the queue's binding key arrives at the exchange?